### ISSUE TITLE
Starting engine error: could not create drawing context

### ISSUE THREAD

**Issue by McSeem** [OpenXRay/xray-16#373]
While render device is starting, an error has appear:

> Could not create drawing context: The specified window isn't an OpenGL window.

System: Kubuntu 18.04.1 amd64
CPU: AMD Ryzen 7 2700
Video: Vega 64 Sapphire Nitro+
Video driver: amdgpu-pro 18.40

**Comment by eagleivg** [comment]
Please, show you `glxinfo -B` output

**Comment by McSeem** [comment]
My OpenGL info:
```
$ glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Memory info (GL_ATI_meminfo):
    VBO free memory - total: 7052 MB, largest block: 6108 MB
    VBO free aux. memory - total: 31483 MB, largest block: 24128 MB
    Texture free memory - total: 7052 MB, largest block: 6108 MB
    Texture free aux. memory - total: 31483 MB, largest block: 24128 MB
    Renderbuffer free memory - total: 7052 MB, largest block: 6108 MB
    Renderbuffer free aux. memory - total: 31483 MB, largest block: 24128 MB
OpenGL vendor string: Advanced Micro Devices, Inc.
OpenGL renderer string: Radeon RX Vega
OpenGL core profile version string: 4.6.13540 Core Profile Context 18.40.2.15
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.5.13540 Compatibility Profile Context 18.40.2.15
OpenGL shading language version string: 4.50
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: 4.5.13540 Compatibility Profile Context 18.40.2.15
OpenGL ES profile shading language version string: 4.50
```

**Comment by lonelyfootpad** [comment]
I just tried this on my cheap laptop as a test and I am getting the same thing happening. Running Ubuntu 18.10. It's not a gaming laptop or anything like that so I wouldn't have expected good performance but I thought I'd give it a try to confirm.

Edit: I also noticed at the very start of the console output "Couldn't create surface from image: Couldn't open logo.bmp", not sure if it's failing at this point to create a window and therefore can't create the OpenGL context?

**Comment by eagleivg** [comment]
Try running other applications using OpenGL, start with glxgears. If there is the same problem, then the problem is in the driver.

**Comment by lonelyfootpad** [comment]
glxgears runs fine, recording a solid 60FPS from it. The Unigine Heaven benchmark also runs fine. It's a crappy laptop so the performance sucks but it runs and renders fine.

![Screenshot from 2019-03-24 06-06-42](https://user-images.githubusercontent.com/14933532/54871162-e07e1480-4dfb-11e9-9fd0-d9d9486bf53d.png)
![Screenshot from 2019-03-24 06-08-45](https://user-images.githubusercontent.com/14933532/54871163-e07e1480-4dfb-11e9-990c-be8849bdf001.png)

**Comment by lonelyfootpad** [comment]
> Edit: I also noticed at the very start of the console output "Couldn't create surface from image: Couldn't open logo.bmp", not sure if it's failing at this point to create a window and therefore can't create the OpenGL context?

I fixed this by exporting the logo .psd file as logo.bmp and putting it in the bin directory with the xr_3da.sh script and it now momentarily shows up on launch but still segfaults right away with 
"Starting RENDER device...
Could not create drawing context: The specified window isn't an OpenGL window"
being the last output before the fatal error and stack trace.

**Comment by eagleivg** [comment]
Ok, let`s try this solution: find and comment in src/xrEngine/Device_Initialize.cpp next lines
```
            SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 4);
            SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 1);
```
and do rebuiild.

**Comment by lonelyfootpad** [comment]
> Ok, let`s try this solution: find and comment in src/xrEngine/Device_Initialize.cpp next lines
> 
> ```
>             SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 4);
>             SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 1);
> ```
> 
> and do rebuiild.

Done and still getting the same thing happening.

**Comment by apocsurvivor** [comment]
The actual error seems to be from src/Layers/xrRenderGL/glHW.cpp line 49, and is called somewhere after src/xrEngine/Device_create.cpp line 39 but before the end of that function (void CRenderDevice::Create() )

I am not a C/C++ developer so I am not really sure what I am looking at. I wish I could be more help.

**Comment by lonelyfootpad** [comment]
I tried commenting out the if statement in src/xrEngine/Device_Initialize.cpp line 50
if (psDeviceFlags.test(rsRGL))
which seems to be doing some sort of check to see if it should set the openGL flags for the SDL window... after removing that check to force the opengl window hints and re-compiling the game starts up. The videos play and the main menu is functional. Tried starting a new game and got a green-screen where the intro videos would normally play and then the familiar new game location. So it seems to be working.

It seems that check it is doing to see if it should include the openGL context stuff is failing and causing those openGL window hints not to be added.

**Comment by McSeem** [comment]
> I tried commenting out the if statement in src/xrEngine/Device_Initialize.cpp line 50
> if (psDeviceFlags.test(rsRGL))

This also fixes a problem with "drawing context" error on my machine - the launch phase of render device is fine now:
```
Starting RENDER device...
* GPU vendor: [ATI Technologies Inc.] device: [Radeon RX Vega]
* GPU OpenGL version: 4.1.13540 Core Profile Context 18.40.2.15
* GPU OpenGL shading language version: 4.50
* GPU OpenGL VTF units: [32] CTI units: [160]
* DVB created: 4096K
* DIB created: 512K
```
But I get a new error:
```
FATAL ERROR
 
[error] Expression    : folder
[error] Function      : LS_Load
[error] File          : xrRenderPC_GL/glResourceManager_Scripting.cpp
[error] Line          : 417
[error] Description   : Shader folder not found: 
[error] Arguments     : gl\
```
Maybe something is wrong with game data? I use the data, that has been installed from original Game DVD. Game resources are packed in *.db files of "resources" directory.

**Comment by lonelyfootpad** [comment]
> Maybe something is wrong with game data? I use the data, that has been installed from original Game DVD. Game resources are packed in *.db files of "resources" directory.

Did you copy the files from the **res** directory across into the directory the game files are in?

**Comment by McSeem** [comment]
> Did you copy the files from the res directory across into the directory the game files are in?

I copied all files from "resources" directory up a level to the root and into "bin-linux-dbg", where built game binaries are in. The same error.

**Comment by Xottab-DUTY** [comment]
> I copied all files from "resources"

Did you copy files from the **`res`** folder? This folder is located in our repository, you don't need to do anything with `resources` folder because it's from original game.

Please, don't mix `res` and `resources`. That are different folders.

**Comment by McSeem** [comment]
> Please, don't mix res and resources. That are different folders.

I copied files from "res" that is in the repository. During startup, the screen resolution changes to 1024х768 and I get this error:
```
Starting RENDER device...
* GPU vendor: [ATI Technologies Inc.] device: [Radeon RX Vega]
* GPU OpenGL version: 4.1.13540 Core Profile Context 18.40.2.15
* GPU OpenGL shading language version: 4.50
* GPU OpenGL VTF units: [32] CTI units: [160]
* DVB created: 4096K
* DIB created: 512K
! Renderer doesn't support blender 'effects\shadow_world'
Starting engine...
start new thread [X-RAY Primary thread]
start new thread [X-RAY Secondary thread]
 
FATAL ERROR
 
[error] Expression    : shader
[error] Function      : shader_compile
[error] File          : xrRenderPC_GL/rgl_shaders.cpp
[error] Line          : 610
[error] Description   : assertion failed
```

**Comment by lonelyfootpad** [comment]
> 
> 
> > Please, don't mix res and resources. That are different folders.
> 
> I copied files from "res" that is in the repository. During startup, the screen resolution changes to 1024х768 and I get this error:
> 
> ```
> Starting RENDER device...
> * GPU vendor: [ATI Technologies Inc.] device: [Radeon RX Vega]
> * GPU OpenGL version: 4.1.13540 Core Profile Context 18.40.2.15
> * GPU OpenGL shading language version: 4.50
> * GPU OpenGL VTF units: [32] CTI units: [160]
> * DVB created: 4096K
> * DIB created: 512K
> ! Renderer doesn't support blender 'effects\shadow_world'
> Starting engine...
> start new thread [X-RAY Primary thread]
> start new thread [X-RAY Secondary thread]
>  
> FATAL ERROR
>  
> [error] Expression    : shader
> [error] Function      : shader_compile
> [error] File          : xrRenderPC_GL/rgl_shaders.cpp
> [error] Line          : 610
> [error] Description   : assertion failed
> ```

Seems like that is a separate issue now.

**Comment by McSeem** [comment]
> Seems like that is a separate issue now.

I'm not sure. I did everything again and this bug did not happen, the game starts fine. However, without removing the "if" statement, it is still not built.

**Comment by Xottab-DUTY** [comment]
```FATAL ERROR
 
[error] Expression    : shader
[error] Function      : shader_compile
[error] File          : xrRenderPC_GL/rgl_shaders.cpp
[error] Line          : 610
[error] Description   : assertion failed
```
@McSeem, please tell us if that shader compilation error will appear again.

Closing the issue, since it was fixed in 59e1df8dbb32e53797f7185ae695b0378ffeba26
